### PR TITLE
Spring Cloud Square

### PR DESCRIPTION
**Pull request by joshlong**
This PR adds support for building Spring Cloud Square-powered OkHttp and Retrofit-based HTTP clients, optionally also supporting client-side loadbalancing.

### CODE REVIEW

File: samples/square/pom.xml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    <description>Demo project for Spring Boot</description>-->
+
+    <parent>
+        <groupId>org.springframework.experimental</groupId>
+        <artifactId>spring-native-sample-parent</artifactId>
+        <version>0.10.2-SNAPSHOT</version>
+        <relativePath>../maven-parent/pom.xml</relativePath>
+    </parent>
+    <groupId>com.example</groupId>
+    <artifactId>async</artifactId>
```
Comment (sdeleuze) [inline samples/square/pom.xml:26]: Could we change that to be = to the directory name "square"?

File: samples/square/pom.xml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    <version>0.0.1-SNAPSHOT</version>
+    <properties>
+        <java.version>1.8</java.version>
+        <spring-cloud.version>2020.0.3</spring-cloud.version>
+
+        <!--
+        TODO switch this to M2 when its released.
+        -->
+        <spring-cloud-square.version>
+            0.4.0-SNAPSHOT
```
Comment (sdeleuze) [inline samples/square/pom.xml:36]: Let's wait `0.4.0-M2` to merge this PR.

**Review by sdeleuze — CHANGES_REQUESTED** [review]
Thanks for this contribution, I added a few comments + could you please check why the PR CI build is broken?

File: ci/pipeline.yml [hunk 3]
```diff
@@ -128,11 +128,6 @@ resources:
       repository: spring-native
```
Comment (aclement) [inline ci/pipeline.yml:128]: why does the PR modify our ci pipeline?

File: spring-native-configuration/src/main/java/org/springframework/cloud/square/SquareHints.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	*
+	* @author Josh Long
+	* @author Andy Clement
+	*/
+@NativeHint(
+	trigger = RetrofitAutoConfiguration.class,
+	options = "-H:+AddAllCharsets --enable-url-protocols=http,https",
+	types = {
+		@TypeHint(typeNames = {
+			"jdk.vm.ci.meta.JavaKind$FormatWithToString[]",
```
Comment (aclement) [inline spring-native-configuration/src/main/java/org/springframework/cloud/square/SquareHints.java:26]: @sdeleuze I recall this one was weird but I thought it was fixed by something in GraalVM wasn't it?

File: spring-native-configuration/src/main/java/org/springframework/cloud/square/SquareHints.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+			"java.lang.reflect.AnnotatedElement[]",
+			"java.lang.reflect.GenericDeclaration[]",
+			"com.oracle.svm.core.hub.Target_java_lang_constant_Constable[]",
```
Comment (aclement) [inline spring-native-configuration/src/main/java/org/springframework/cloud/square/SquareHints.java:29]: These 3 are duplicates are what you have in the square sample project reflect-config.json - feels weird to have them in both places

File: spring-native-configuration/src/main/java/org/springframework/cloud/square/SquareHints.java (hunk 4 above)
Comment (sdeleuze) [reply spring-native-configuration/src/main/java/org/springframework/cloud/square/SquareHints.java:26]: I am not sure, we still have it in `LoadBalancerHints`, likely to be tested to check.

File: ci/pipeline.yml (hunk 3 above)
Comment (joshlong) [reply ci/pipeline.yml:128]: no earthly idea. please feel free to ignore any changes to the CI pipeline

**Comment by aclement** [comment]
I've polished this up a bit in https://github.com/spring-projects-experimental/spring-native/pull/945 - but that won't go in until M2 of square is out (because it doesn't work for M1 and I'd rather not have a snapshot dependency there), we may hold it off until 0.11 as well because that changes many things.

**Pull request closed without merging**